lib/src/player.dart

```diff
@@ -143,7 +154,7 @@ class Player with PlayerEventHandler implements PlayerApi {
   ]) async {
     final result = await _initializationResult;
     if (!result) {
-      return Future.error('Error initializing player on native platform side.');
```

Comment:
Wrapping in `Future.error` is unnecessary in an async function, it is equivalent to `throw`.
Additionally, the code was throwing a string when Dart recommends to only throw subclasses of Exception or Error.